pkg/model/manifest.go

```diff
@@ -0,0 +1,356 @@
+package model
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/uptrace/bun"
+	"github.com/veraison/corim/comid"
+	"github.com/veraison/corim/corim"
+	"github.com/veraison/eat"
+	"github.com/veraison/swid"
+)
+
+type ManifestIDType string
+
+type ProfileType string
+
+type Manifest struct {
+	bun.BaseModel `bun:"table:manifests,alias:man"`
+
+	ID int64 `bun:",pk,autoincrement"`
+
+	ManifestIDType TagIDType
+	ManifestID     string
+
+	Digest    []byte
+	TimeAdded time.Time
+	Label     string `bun:",nullzero"`
+
+	ProfileType ProfileType `bun:",nullzero"`
+	Profile     string      `bun:",nullzero"`
+
+	Entities      []*Entity  `bun:"rel:has-many,join:id=owner_id,join:type=owner_type,polymorphic:manifest"`
+	DependentRIMs []*Locator `bun:"rel:has-many,join:id=manifest_id"`
+
+	NotBefore *time.Time
+	NotAfter  *time.Time
+
+	ModuleTags []*ModuleTag `bun:"rel:has-many,join:id=manifest_id"`
```

Comment:
A question on line 43: Should we not call it as CoMIDTag, so that in future, if need to insert, CoTS or CoTL, this will fit in nicely..?